src/fsharp/fsc.fs

```diff
@@ -545,18 +546,25 @@ let main1(ctok, argv, legacyReferenceResolver, bannerAlreadyPrinted,
 
     let inputs =
         try
-            let isLastCompiland, isExe = sourceFiles |> tcConfig.ComputeCanContainEntryPoint 
-
-            List.zip sourceFiles isLastCompiland
-            // PERF: consider making this parallel, once uses of global state relevant to parsing are cleaned up 
-            |> List.choose (fun (sourceFile, isLastCompiland) -> 
-
-                let sourceFileDirectory = Path.GetDirectoryName sourceFile
-
-                match ParseOneInputFile(tcConfig, lexResourceManager, ["COMPILED"], sourceFile, (isLastCompiland, isExe), errorLogger, (*retryLocked*)false) with
-                | Some input -> Some (input, sourceFileDirectory)
-                | None -> None) 
-
+            let isLastCompiland, isExe = sourceFiles |> tcConfig.ComputeCanContainEntryPoint
+            let sourceFiles = isLastCompiland |> List.zip sourceFiles |> Array.ofSeq
+            let parallelOptions = ParallelOptions(MaxDegreeOfParallelism=min Environment.ProcessorCount sourceFiles.Length)
+
+            let results = Array.zeroCreate sourceFiles.Length
+            Parallel.For(0, sourceFiles.Length, parallelOptions, fun i ->
+                results.[i] <-
+                    let (filename: string, isLastCompiland) = sourceFiles.[i]
+                    let pathOfMetaCommandSource = Path.GetDirectoryName filename
+                    let delayedErrorLogger = errorLoggerProvider.CreateDelayAndForwardLogger(exiter)
+                    match ParseOneInputFile(tcConfig, lexResourceManager, ["COMPILED"], filename, (isLastCompiland, isExe), delayedErrorLogger, (*retryLocked*)false) with
+                    | Some input -> delayedErrorLogger, Some (input, pathOfMetaCommandSource)
+                    | None -> delayedErrorLogger, None) |> ignore
+            results
+            |> Array.choose (fun (delayedErrorLogger, result) -> 
```

Comment:
can `Array.Parallel.choose` be used here?

Reply:
I don't want to use parallel choose as I want to commit the logger results synchronously.